ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -356,6 +360,14 @@ protected String getUuid() {
         idBuf.putShort(accountId);
         idBuf.putShort(containerId);
         break;
+      case BLOB_ID_V3:
```

Comment:
minor: Can you keep the switch case for the versions in order?

Reply:
Perhaps this can be shortened as follows:
```java
{
  ...
  flag = (byte) 0;
  switch(version) {
  case BLOB_ID_V4:
    flag |= isEncrypted ? IS_ENCRYPTED_MASK : 0;
    // fall through
  case BLOB_ID_V3:
    flag = type.ordinal() & BLOB_ID_TYPE_MASK;
    // fall through
  case BLOB_ID_V2:
    idBuf.put(flag);
    ...
    idBuf.putShort(containerId);
    // fall through (or break);
   case BLOB_ID_V1:
     break;
   default:
     throw;
  }
  idBuf.put(partitionId.getBytes());
  ...
  idBuf.put(uuid.getBytes());
  return idBuf.array();
}
```
There is a pattern that this reflects (earlier versions being subsumed under later ones). But if you find it less readable, let us have separate switch-cases (instead of the pattern-less combining of just a few fields).

Reply:
Looks like versions in order have better readability.